### PR TITLE
test(atomic): Add cypress test for category facet search and basepath

### PR DESCRIPTION
**Pull request by lvu285**
- Add check for last level item is not a button 
- Add some tests for basePath
- Add test for Category Facet search Path

** some tests for highlighted & UA, I disabled them for now, as we still have some small issues with that

### CODE REVIEW

File: packages/atomic/cypress/integration/facets/category-facet.cypress.ts [hunk 1]
```diff
@@ -483,3 +548,76 @@ describe('When URL contains a selected path of category facet', () => {
...
+  });
+
+  it('Category facet should load properly', () => {
+    validateFacetComponentLoaded(
+      categoryFacetProp.label,
+      FacetSelectors.categoryFacet
+    );
+  });
+
+  it('Should load and not filter results based on the basePath', () => {
```
Comment (samisayegh) [inline packages/atomic/cypress/integration/facets/category-facet.cypress.ts:612]: If there is a basePath, should this description say `should load and filter`?

**Review by samisayegh — APPROVED** [review]

File: packages/atomic/cypress/integration/facets/facet-utils.ts [hunk 2]
```diff
@@ -279,3 +279,22 @@ export function assertShowLessUA(field: string) {
     expect(analyticsBody.customData).to.have.property('facetField', field);
   });
 }
+
+export function typeQueryAndWaitUA(searchboxSelector: string, query: string) {
```
Comment (ThibodeauJF) [inline packages/atomic/cypress/integration/facets/facet-utils.ts:283]: could we have a new facet-search-utils file for these?

Comment (lvu285) [reply packages/atomic/cypress/integration/facets/facet-utils.ts:285]: facetSearch is/ and will be included in same test suites of Facets, and so far I don't see if we will have much function for that particular facet-search, so I suggest keep like this for now :)

File: packages/atomic/cypress/integration/facets/category-facet.cypress.ts [hunk 3]
```diff
@@ -347,9 +361,61 @@ describe('Category Facet with facetSearchEnable', () => {
       .should('be.visible');
   });
 
-  describe('When user starts typing on facet searchbox', () => {
-    it('Should display search results correctly');
-    it('Should trigger new filter when item is selected from search dropdown');
+  describe.skip('When user interacts with category facet searchbox', () => {
```
Comment (ThibodeauJF) [inline packages/atomic/cypress/integration/facets/category-facet.cypress.ts:364]: This  should work now

**Review by ThibodeauJF — APPROVED** [review]

File: packages/atomic/cypress/integration/facets/category-facet.cypress.ts (hunk 3 above)
Comment (lvu285) [reply packages/atomic/cypress/integration/facets/category-facet.cypress.ts:364]: oh good to know, I will double check, if it works, I will enable it in this PR as well

**Pull request merged**